packages/near-api-js/src/providers/wallet-rpc-provider.ts

```diff
@@ -0,0 +1,473 @@
+/**
+ * @module
+ * @description
+ * This module contains the {@link JsonRpcProvider} client class
+ * which can be used to interact with the [NEAR RPC API](https://docs.near.org/api/rpc/introduction).
+ * @see {@link providers/provider | providers} for a list of request and response types
+ */
+import {
+    AccessKeyWithPublicKey,
+    Provider,
+    FinalExecutionOutcome,
+    NodeStatusResult,
+    BlockId,
+    BlockReference,
+    BlockResult,
+    BlockChangeResult,
+    ChangeResult,
+    ChunkId,
+    ChunkResult,
+    EpochValidatorInfo,
+    NearProtocolConfig,
+    LightClientProof,
+    LightClientProofRequest,
+    GasPrice,
+    QueryResponseKind
+} from './provider';
+import { TypedError, ErrorContext } from '../utils/errors';
+import { baseEncode } from 'borsh';
+import exponentialBackoff from '../utils/exponential-backoff';
+import { parseRpcError } from '../utils/rpc_errors';
+import { SignedTransaction, Transaction } from '../transaction';
+
+/** @hidden */
+export { TypedError, ErrorContext };
+
+// Default number of retries before giving up on a request.
+const REQUEST_RETRY_NUMBER = 12;
+
+// Default wait until next retry in millis.
+const REQUEST_RETRY_WAIT = 500;
+
+// Exponential back off for waiting to retry.
+const REQUEST_RETRY_WAIT_BACKOFF = 1.5;
```

Comment:
How were these retry policy variables determined? 12 retries seems a bit high, what is total time of before request fails?